Tensile/SolutionStructs.py

```diff
@@ -2482,15 +2484,11 @@ def isDirectToLdsDoable(state, tc):
       reject(state, "DirectToLds can only be used with buffer loads requiring 1 register")
       return False
 
-    if state["ProblemType"]["DataType"].isHalf():
-      if state["AssertSummationElementMultiple"] % (2 * state["GlobalLoadVectorWidth%c"%tc])  != 0:
-        reject(state, "can't use DirectToLds for FP16 with AssertSummationElementMultiple %u" % state["AssertSummationElementMultiple"])
+    if (state["ProblemType"]["DataType"].isHalf() or state["ProblemType"]["DataType"].isBFloat16()) and state["ProblemType"]["TLU%c"%tc] == False:
+      if (asem % gsu != 0) or ((asem//gsu) % (state["GlobalLoadVectorWidth%c"%tc])  != 0):
+        reject(state, "can't use DirectToLds for FP16/BF16 with TLU%c=False and AssertSummationElementMultiple=%u and GlobalSplitU=%u" % (tc, asem, gsu))
```

Comment:
Do the changes only apply to half? Do we support kernels where asem % gsu != 0 ?

Reply:
This part tries to guarantee K is multiple of 2 for half.
For SGEMM (or other types), it does not matter if K is multiple of something.
ASEM check should be necessary only for half + TLU = False.
(half + TLU=True case, ASEM does not matter)